### PR TITLE
[NewPM][CodeGen] Port VirtRegMap to NPM

### PR DESCRIPTION
**Pull request by optimisan**

### CODE REVIEW

**Comment by Akshat-Oke** [comment]
* **#109939** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/109939?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#109938** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/109938?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#109937** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/109937?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#109936** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/109936?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Akshat-Oke and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: llvm/include/llvm/CodeGen/VirtRegMap.h [hunk 1]
```diff
@@ -30,171 +31,210 @@ class MachineRegisterInfo;
...
+inline raw_ostream &operator<<(raw_ostream &OS, const VirtRegMap &VRM) {
+  VRM.print(OS);
+  return OS;
+}
 
-    /// returns true if VirtReg has a known preferred register.
-    /// This returns false if VirtReg has a preference that is a virtual
-    /// register that hasn't been assigned yet.
-    bool hasKnownPreference(Register VirtReg) const;
+class VirtRegMapWrapperPass : public MachineFunctionPass {
```
Comment (arsenm) [inline llvm/include/llvm/CodeGen/VirtRegMap.h:193]: WrapperLegacy to avoid PassPass

Comment (cdevadas) [reply llvm/include/llvm/CodeGen/VirtRegMap.h:193]: Unfortunately, most of the analysis passes when ported to NPM had ended up with PassPass in the initialize* functions. We should have followed 'WrapperLegacy' everywhere.

**Review by arsenm — APPROVED** [review]

**Comment by optimisan** [comment]
On hold, something's wrong.

**Review by arsenm — APPROVED** [review]

**Comment by optimisan** [comment]
### Merge activity

* **Oct 22, 5:41 AM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/llvm/llvm-project/109936).

**Pull request merged**